metricflow/test/sql_clients/test_sql_client.py

```diff
@@ -128,3 +134,30 @@ def test_dry_run_of_bad_query_raises_exception(sql_client: SqlClient) -> None:
     # Tests that a bad query raises an exception, also tests that EXPLAIN gets prepended to the stmt
     with pytest.raises(OperationalError, match=rf"EXPLAIN {bad_stmt}"):
         sql_client.dry_run(bad_stmt)
+
+
+def test_cancel_submitted_queries(  # noqa: D
+    request: FixtureRequest, mf_test_session_state: MetricFlowTestSessionState, sql_client: SqlClient
+) -> None:
+    if (
+        not sql_client.sql_engine_attributes.sleep_supported
+        or not sql_client.sql_engine_attributes.cancel_submitted_queries_supported
+    ):
+        logger.info(
+            f"Skipping test {request.node.fspath}::{request.node.name} since the sql_client does not support features "
+            f"required for this test."
+        )
+        return
```

Comment:
Hm. Maybe call `pytest.skip()` here instead of a logger.info() and return statement? That way the results show a skip rather than a fake pass for engines that don't support this - having code that "passes" in the relevant target tests only to find out much later that it was being bypassed is a deeply frustrating dev experience.

Reply:
The issue with the skip annotation is that it can't make use of the built in fixtures.

Reply:
We could make that happen with a global variable, but that doesn't seem great.

Reply:
Actually, it seems like `pytest.xfail` can help accomplish similar behavior.

Reply:
You can use pytest.skip inside the method. It's just a function call, behaves just like the decorator but it's invoked imperatively.

If the xfail decorator will work outside the function that's way better though.

Reply:
Nice, didn't realize you could call `pytest.skip` inside the method. I had seen docs for using `pytest.xfail` as a function call, so I was referring to doing that. I haven't seen a way of using `pytest.xfail` as a decorator for this scenario.